syftlib/src/main/java/org/openmined/syft/execution/JobDownloader.kt

```diff
@@ -0,0 +1,181 @@
+package org.openmined.syft.execution
+
+import android.util.Log
+import io.reactivex.Single
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.processors.PublishProcessor
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.networking.datamodels.ClientConfig
+import org.openmined.syft.proto.SyftModel
+import org.openmined.syft.utilities.FileWriter
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicReference
+
+private const val TAG = "JobDownloader"
+
+@ExperimentalUnsignedTypes
+internal class JobDownloader {
```

Comment:
This is a first step in the refactoring applying SRP.
This class and its methods can (and have to) be improved in the following iterations.